aspnetcore/host-and-deploy/health-checks/samples/2.x/HealthChecksSample/Services/StartupHostedService.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace SampleApp.Services
+{
+    #region snippet1
+    public class StartupHostedService : IHostedService, IDisposable
+    {
+        private readonly int _delaySeconds = 15;
+        private readonly ILogger _logger;
+        private readonly StartupHostedServiceHealthCheck _startupHostedServiceHealthCheck;
+
+        public StartupHostedService(ILogger<StartupHostedService> logger, 
+            StartupHostedServiceHealthCheck startupHostedServiceHealthCheck)
+        {
+            _logger = logger;
+            _startupHostedServiceHealthCheck = startupHostedServiceHealthCheck;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
```

Comment:
I think it's worth a comment in here that this code exists for sample purposes, and to **simulate** the effect of a long-running startup job.